pkg/flow/module.go

```diff
@@ -136,7 +138,6 @@ func (c *module) LoadConfig(config []byte, args map[string]any) error {
 //
 // Run blocks until the provided context is canceled.
 func (c *module) Run(ctx context.Context) {
-	defer c.o.parent.removeID(c.o.ID)
```

Comment:
IMO I think we should keep this line; it's a good safety precaution to remove modules once they stop running.